clang/test/Driver/sycl-oneapi-gpu.cpp

```diff
@@ -87,6 +87,101 @@
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx803 -DMAC_STR=GFX803
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx805 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx805 -DMAC_STR=GFX805
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx810 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx810 -DMAC_STR=GFX810
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx900 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx900 -DMAC_STR=GFX900
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx902 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx902 -DMAC_STR=GFX902
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx904 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx904 -DMAC_STR=GFX904
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx906 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx906 -DMAC_STR=GFX906
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx908 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx908 -DMAC_STR=GFX908
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx90a -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx90a -DMAC_STR=GFX90A
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1010 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1010 -DMAC_STR=GFX1010
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1011 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1011 -DMAC_STR=GFX1011
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1012 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1012 -DMAC_STR=GFX1012
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1013 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1013 -DMAC_STR=GFX1013
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1030 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1030 -DMAC_STR=GFX1030
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1031 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1031 -DMAC_STR=GFX1031
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx1032 -### %s 2>&1 | \
+// RUN:   FileCheck %s --check-prefixes=DEVICE_AMD,MACRO_AMD -DDEV_STR=gfx1032 -DMAC_STR=GFX1032
+// MACRO_AMD: clang{{.*}} "-triple" "amdgcn-amd-amdhsa"
+// MACRO_AMD: "-D__SYCL_TARGET_AMD_GPU_[[MAC_STR]]__"
+// DEVICE_AMD: clang-offload-wrapper{{.*}} "-compile-opts=--offload-arch=[[DEV_STR]]"
+// MACRO_AMD: clang{{.*}} "-fsycl-is-host"
+// MACRO_AMD: "-D__SYCL_TARGET_AMD_GPU_[[MAC_STR]]__"
+
```

Comment:
Below the test checks for input validation, I think this need to be extended to cover cuda/amd and for miss match like `nvidia_gpu_skl`.

There is also a check for phases (last one), would be good to have one to make sure arch is properly propagated.